pkgs/development/python-modules/hydra-check/default.nix

```diff
@@ -0,0 +1,50 @@
+{ lib
+, buildPythonPackage
+, pythonOlder
+, fetchFromGitHub
+, docopt
+, requests
+, beautifulsoup4
+, black
+, mypy
+, flake8
+}:
+
+buildPythonPackage rec {
+  pname = "hydra-check";
+  version = "1.1.1";
+  disabled = pythonOlder "3.5";
+
+  src = fetchFromGitHub {
+    owner = "nix-community";
+    repo = "hydra-check";
+    rev = version;
+    sha256 = "1dmsscsib8ckp496gsfqxmq8d35zs71n99xmziq9iprvy7n5clq2";
+
+  };
+
+  propagatedBuildInputs = [
+    docopt
+    requests
+    beautifulsoup4
+  ];
+
+  checkInputs = [ black mypy flake8 ];
+
+  checkPhase = ''
+    echo -e "\x1b[32m## run black\x1b[0m"
+    LC_ALL=C black --check $(find -name \*.py)
+    echo -e "\x1b[32m## run flake8\x1b[0m"
+    flake8 hydracheck
+    echo -e "\x1b[32m## run mypy\x1b[0m"
+    mypy hydracheck
+  '';
```

Comment:
i agree with black and flake8 but mypy seems to also check type hints. It might be useful to include it in case something changes with the next python version, no?

Reply:
I think that's fair. But mypy has also know to break some APIs, although I can't think of any in particular.

Reply:
or maybe it was that it became more strict. Either way, I've seen where a bump in mypy has caused failures